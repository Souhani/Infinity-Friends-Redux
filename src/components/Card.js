const Card = ({filteredRobots , onDeleteRobot, selectedAvatars }) => {
    return(
        <div>
            {
                filteredRobots.map(
                    (card, key) => {
                        let felterRobots;
                        return(
                            <div className='bg-light-yellow  tc br3 dib shadow-2 dib pa3 bw2 ma2'
                             key={key+card.id}
                                style={{position: 'relative'}}>
                                <button className="b ph2  grow link  bg-gold pv1  br4 ba b--yellow pointer f6 "
                                        onClick={ felterRobots =() => {onDeleteRobot(card.lastName)}}
                                        style={{position: 'absolute', top:'1%', right:'1%'}}>
                                        Delete
                                </button>
                                <div className="pa3">
                                    <img 
                                        alt='robot'
                                        src= {`https://robohash.org/${card.firstName+card.id+card.lastName}?set=${selectedAvatars}&size=100x100`} />
                                    </div>
                                <div className='grow navy'
                                     style={{cursor: "pointer"}}>
                                    <h2>{card.firstName}</h2>
                                    <h4>{card.lastName}</h4>
                                </div>
                           </div>
                        )   
                    }  
                )
            }      
        </div>     
    )
};
export default Card;
