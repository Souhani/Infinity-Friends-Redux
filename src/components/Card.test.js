import React from 'react';
import { shallow } from 'enzyme';
 
import Card from './Card';

it ('testing card', () => {
    expect.assertions(3);
    const robotsMock = [{
        id:1,
        firstName:'jhon',
        lastName:'snow'
    },];
    const mockHandleDeleteCard = jest.fn();
    const wrapper= shallow(<Card filteredRobots={robotsMock} onDeleteRobot={mockHandleDeleteCard}/>);
    //test rendering
    expect(wrapper).toMatchSnapshot();
    //test clicking
    wrapper.find('button').simulate('click');
    expect(mockHandleDeleteCard.mock.calls.length).toBe(1);
    expect(mockHandleDeleteCard).toBeCalledWith("snow");

})