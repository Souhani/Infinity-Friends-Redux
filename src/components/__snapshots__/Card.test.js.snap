// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing card 1`] = `
<div>
  <div
    className="bg-light-yellow  tc br3 dib shadow-2 dib pa3 bw2 ma2"
    key="1"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <button
      className="b ph2  grow link  bg-gold pv1  br4 ba b--yellow pointer f6 "
      onClick={[Function]}
      style={
        Object {
          "position": "absolute",
          "right": "1%",
          "top": "1%",
        }
      }
    >
      Delete
    </button>
    <img
      alt="robot"
      src="https://robohash.org/jhon1snow?set=undefined&size=100x100"
    />
    <div
      className="grow navy "
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <h2>
        jhon
      </h2>
      <h4>
        snow
      </h4>
    </div>
  </div>
</div>
`;
